404-server/metrics.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Collect and display prometheus metrics
+
+package main
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	namespace = "default_http_backend"
+	subsystem = "http"
+)
+
+var (
+	requestCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: namespace,
+		Subsystem: subsystem,
+		Name:      "request_count_total",
+		Help:      "Counter of HTTP requests made.",
+	}, []string{"proto"})
+
+	requestDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: namespace,
+		Subsystem: subsystem,
+		Name:      "request_duration_milliseconds",
+		Help:      "Histogram of the time (in milliseconds) each request took.",
+		Buckets:   append([]float64{.001, .003}, prometheus.DefBuckets...),
```

Comment:
I did have this at microseconds, but I didn't know what the average response time would be "in the wild". Would you like me to switch to microseconds?